### PR TITLE
math.s

### PR DESCRIPTION
**Pull request by Kermalis**
~~WIP~~

### CODE REVIEW

**Comment by Kermalis** [comment]
Done

File: src/math.c [hunk 1]
```diff
@@ -189,137 +255,689 @@ u24_8 u24_8_div(u24_8 x, u24_8 y)
...
+    aa.s0 = a->s0;
+    aa.s4 = a->s4;
+    bb.s0 = b->s0;
+    bb.s4 = b->s4;
+    aIsNegative = F48_16_IsNegative(&aa);
+    bIsNegative = F48_16_IsNegative(&bb);
+
+    if (F48_16_IsZero(&bb)) {
+        dst->s0 = INT32_MAX;
+        dst->s4 = UINT32_MAX;
```
Comment (mateon1) [inline src/math.c:641]: I'm not sure using *INT32_MAX consts is semantically correct here, I think I'd prefer `~(1<<31)` (or `~0u>>1`) and `~0` actually

File: src/math.c [hunk 2]
```diff
@@ -189,137 +255,689 @@ u24_8 u24_8_div(u24_8 x, u24_8 y)
...
+    u32 r7;
+    u32 r9;
+    s32 sp4;
+    u32 sp8;
+    u32 spC;
+    s32 i;
+
+    if (F48_16_IsZero(b)) {
+        dst->s0 = INT32_MAX;
+        dst->s4 = UINT32_MAX;
```
Comment (mateon1) [inline src/math.c:770]: Same here

File: src/math.c [hunk 3]
```diff
@@ -189,137 +255,689 @@ u24_8 u24_8_div(u24_8 x, u24_8 y)
...
+        sp8 = sp10;
+    }
+
+    if (!F48_16_IsZero(&sp8)) {
+        i = 0;
+        goto deez; // Fakematch cuz ya (https://decomp.me/scratch/2TCrJ)
+
+        while (i != 2) {
+            F48_16_SMul(&sp8, &sp8, &sp10);
+            i++;
```
Comment (mateon1) [inline src/math.c:916]: Could this be a for, with `SMul(...), i++` in the increment?

Comment (Kermalis) [reply src/math.c:916]: Already tried it many times, you can check it https://decomp.me/scratch/2TCrJ

**Comment by AnonymousRandomPerson** [comment]
Merge conflicts from https://github.com/pret/pmd-red/pull/251 merging, as acknowledge in chat.

**Comment by Kermalis** [comment]
We're good now

**Review by AnonymousRandomPerson — APPROVED** [review]

**Comment by AnonymousRandomPerson** [comment]
Will wait on resolution of @mateon1 's comments before merging.

**Pull request merged**